# ML-03 — Frame Your Lane as an ML Task

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane as an ML task (type)

*Classification, clustering, ranking, or scoring — which one, and why?*

My lane is Refresh / Content Opportunity Scoring, framed as a binary classification task.

The model would predict whether a content item is declining or not declining. Although the target is binary, the model would produce a probability for each content item. This probability can be used as a priority score to rank the pages that should be reviewed first.and

Classification is therefore the underlying ML task, while scoring and ranking are how the output supports the content decision.

## 2. Target or proxy

*What would you predict? Where does that label come from — observed outcome or a defined rule?*

The target us **is_declining_labal**.

A value of **1** means that the observed **trend_direction** is down. In the contrary, a value of **0** means that the observed trend direction is not down.

This label is a defined proxy, created from a measured trend rather than from a direct observation that a page truly needs to be refreshed. Decline is treated as a signal that a page may deserve review, but it does not prove that refreshing the page is the correct action.

In [1]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")
df["is_declining_label"] = (df["trend_direction"].astype(str).str.lower() == "down").astype(int)
df["is_declining_label"].value_counts().sort_index()

is_declining_label
0    13738
1    16262
Name: count, dtype: int64

In [ ]:
df["is_declining_label"].value_counts(normalize=True).sort_index() * 100 #to have the percentage 

is_declining_label
0    45.793333
1    54.206667
Name: proportion, dtype: float64

## 3. Success metric

*One metric you can defend. What number means 'good'?*

The succes metric used here is Precision@50.

It measures the proportion of declining content items among the first 50 items recommended for review. This matches the real decision because a content specialist has limited review time and will mainly act on the top recs.

The observed prevalence of the proxy label is 54.2%, so a non-informative ranking would eontain approximately 27 declining items in its top 50.

In [5]:
declining_rate = df["is_declining_label"].mean()
expected_top_50 = declining_rate * 50

print(f"Observed declining rate : {declining_rate:.1%}")
print("Expected declining items in a non-informative top 50:", round(expected_top_50, 1))


Observed declining rate : 54.2%
Expected declining items in a non-informative top 50: 27.1


## 4. The unit of analysis, as a real dataframe

*Load your lane's slice and show it: one row = one what?*

In [6]:
safe = ["trend_direction", "is_declining_label"]
lane = df[safe].copy()
lane.head(10)

,trend_direction,is_declining_label
0,down,1
1,down,1
2,down,1
3,stable,0
4,down,1
5,down,1
6,down,1
7,stable,0
8,down,1
9,down,1


One row represents one anonymized content item, such as one page that could be reviewed. Each row contains measured information about that content item. The target column, **is_declining_label**، indicates whether its observed trend direction is down. Sensitive informations about the clients are not displayed.

## 5. Why ML beats a fixed rule here

*What makes the pattern too messy for an if-statement?*

A fixed rule is enough to create the proxy label: if trend_direction is "down", the content item is labelled as declining.

However, this rule does not show which declining pages represent the strongest review opportunities. Two pages may both be declining, but one may be more important because of its traffic, previous performance or search visibility.

A ML model could combine these different signals to produce a decision-support score and prioritize the pages that should be reviewed first. The model would not prove that refreshing a page will improve its performance, but it could help make the review process more focused.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.